crates/stark-felt/src/lib.rs

```diff
@@ -329,6 +572,56 @@ mod arithmetic {
         }
     }
 
+    /// Field subtraction. Never overflows/underflows.
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    impl ops::Sub<Felt> for u64 {
+        type Output = Felt;
+        fn sub(self, rhs: Felt) -> Self::Output {
+            rhs.neg() + self
+        }
+    }
+
+    /// Field subtraction. Never overflows/underflows.
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    impl ops::Sub<&Felt> for u64 {
+        type Output = Felt;
+        fn sub(self, rhs: &Felt) -> Self::Output {
+            rhs.neg() + self
+        }
+    }
```

Comment:
Yeah, that sounds strange. The first impl is needed that way because it's used to compute relocatable offsets and operating fully on `Felt` became a bottleneck in the VM's main loop. I'm not sure what the `Sub` here is used for, but it sounds likely to also be meant to produce an `u64` (and thus return `Option<u64>`).